### PR TITLE
[MIG] server_environment: Migration to 13.0

### PR DESCRIPTION
**Pull request by grindtildeath**

### CODE REVIEW

File: server_environment/serv_config.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    _description = 'Display server configuration'
+    _conf_defaults = _Defaults()
+
+    @classmethod
+    def _build_model(cls, pool, cr):
+        """Add columns to model dynamically
+        and init some properties
+
+        """
+        ModelClass = super(ServerConfiguration, cls)._build_model(pool, cr)
```
Comment (Tonow-c2c) [inline server_environment/serv_config.py]:
```suggestion
        ModelClass = super()._build_model(pool, cr)
```

File: server_environment/serv_config.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        arch += '<separator colspan="4"/></page>'
+
+        arch += '</notebook></form>'
+        cls._arch = etree.fromstring(arch)
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        """Overwrite the default method to render the custom view."""
+        res = super(ServerConfiguration, self).fields_view_get(view_id,
```
Comment (Tonow-c2c) [inline server_environment/serv_config.py]:
```suggestion
        res = super().fields_view_get(view_id, view_type, toolbar)
```

**Review by Tonow-c2c — APPROVED** [review]

File: server_environment/__init__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from . import models
+# TODO when migrating to 12, fix the import of serv_config by renaming the
+# file?
+# Add an alias to access to the 'serv_config' module as it is shadowed
+# in the following line by an import of a variable with the same name.
+# We can't change the import of serv_config for backward compatibility.
```
Comment (gurneyalex) [inline server_environment/__init__.py]: @grindtildeath do we want to take care of this in the migration to 13.0?

File: server_environment/models/server_env_mixin.py [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+    def _compute_server_env_from_default(self, field_name, options):
+        if options and options.get('compute_default'):
+            getattr(self, options['compute_default'])()
+        else:
+            default_field = self._server_env_default_fieldname(
+                field_name
+            )
+            if default_field:
+                self[field_name] = self[default_field]
+
```
Comment (grindtildeath) [inline server_environment/models/server_env_mixin.py:238]:
```suggestion
            else:
                self[field_name] = False
```

File: server_environment/models/server_env_mixin.py [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+        if options and options.get('compute_default'):
+            getattr(self, options['compute_default'])()
+        else:
+            default_field = self._server_env_default_fieldname(
+                field_name
+            )
+            if default_field:
+                self[field_name] = self[default_field]
+            else:
+                self[field_name] = False
```
Comment (grindtildeath) [inline server_environment/models/server_env_mixin.py:237]:
```suggestion
                self[field_name] = False

```

File: server_environment/__init__.py (hunk 3 above)
Comment (grindtildeath) [reply server_environment/__init__.py]: ~Will do it~ Done ;)

**Review by gurneyalex — APPROVED** [review]

**Comment by gurneyalex** [comment]
@grindtildeath can you squash the fixup commits (including the old ones before your own from this migration)?

**Comment by grindtildeath** [comment]
@gurneyalex Done ;)

**Pull request merged**